### PR TITLE
Translation from a given filepath

### PR DESCRIPTION
**Pull request by Ydrasil**

### CODE REVIEW

**Comment by Ydrasil** [comment]
@vinyll Tests are passing :wink:

File: test/i18n.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
 from unittest import TestCase
-from tcii18n import trans
+from tcii18n.translator import Translator
 
 
 class I18nTest(TestCase):
     def setUp(self):
-        self.trans = trans._
-        self.trans_it = None
+        _ = Translator('test/fixtures.fr.csv').translate
```
Comment (vinyll) [inline test/i18n.py]: self.trans = `Translator('test/fixtures.fr.csv').translate`

File: test/i18n.py [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+        self.trans = _
+        _ = Translator('test/fixtures.it.csv').translate
+        self.trans_it = _
```
Comment (vinyll) [inline test/i18n.py]: same here

**Review by vinyll — CHANGES_REQUESTED** [review]
It seems to be simple and functional 👍 
The only concern I have now is that the csv file is open and parsed every single time we call `translate()` maybe you could store the content in a module variable when a file is run once?

**Comment by Ydrasil** [comment]
@vinyll Changes done, however with this way the file is still open so I have a `ResourceWarning: unclosed file`

File: tcii18n/translator.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import csv
+
+
+class Translator:
+    def __init__(self, filename):
+        self.reader = csv.reader(open(filename, 'r'))
```
Comment (vinyll) [inline tcii18n/translator.py]: you should close your file once it's read.

See [an example here](https://bitbucket.org/anthropedia/tci-data/src/ed0d88c6ed4995ec5195ef60974a1281b2693f67/tcidata/__init__.py?at=master&fileviewer=file-view-default#__init__.py-34)

**Review by vinyll — CHANGES_REQUESTED** [review]

File: tcii18n/translator.py (hunk 3 above)
Comment (Ydrasil) [reply tcii18n/translator.py]: If I close the file when I read it, I have to reopen it every time, like the first commit I did, even in the example, files are opened every time
Or I can stock all the content of the file in a variable

Comment (vinyll) [reply tcii18n/translator.py]: The first solution was opening a file on each call to a translation.
This new version is opening a file only when creating a new instance.
When using it, we probably only want to call the instance once and then call the `translate()` as often as required.

Comment (vinyll) [reply tcii18n/translator.py]: In which case the test should of course be adapted accordingly…

**Comment by Ydrasil** [comment]
@vinyll Now its good :+1:

File: tcii18n/translator.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import csv
+
+
+class Translator:
+    def __init__(self, filename):
+        self.reader = open(filename, 'r')
```
Comment (vinyll) [inline tcii18n/translator.py]: you don't need `reader` as a class property. `reader = …` without self will be enough.

File: test/i18n.py [hunk 5]
```diff
@@ -1,11 +1,11 @@
 from unittest import TestCase
-from tcii18n import trans
+from tcii18n.translator import Translator
 
 
 class I18nTest(TestCase):
     def setUp(self):
-        self.trans = trans._
-        self.trans_it = None
+        self.trans = Translator('test/fixtures.fr.csv').translate
```
Comment (vinyll) [inline test/i18n.py]: for avoiding multiple instance calls as you mentioned in the previous comment, you probably want to instanciate only once:
```
translator = Translator('test/fixtures.fr.csv')
self.trans = translator.translate
```
and similarly for `self.trans_it`

**Review by vinyll — CHANGES_REQUESTED** [review]

**Comment by Ydrasil** [comment]
@vinyll

**Review by vinyll — APPROVED** [review]

**Pull request merged**